index.js

```diff
@@ -106,7 +107,7 @@ const loadPackage = (url, authInfo) => new Promise((resolve, reject) => {
 				reject(e);
 			}
 		});
-	}).on('error', reject);
+	}).on('error', reject).on('timeout', reject);
```

Comment:
Setting `timeout` will only add a `timeout` event to the event emitter.

See https://nodejs.org/api/http.html#http_http_request_options_callback at the bottom:

> Note that setting the timeout option or using the setTimeout() function will not abort the request or do anything besides add a 'timeout' event.